### PR TITLE
Flexible assets

### PR DESCRIPTION
**Pull request by dean1012**
Allows overriding assets location at runtime through the environment variable MWF_DATADIR. If assets.zip is found at that location, it will be mounted. If not, it will try to load the needed files directly from the directory specified.

### CODE REVIEW

**Comment by dean1012** [comment]
@EliteTK Please review this and test on Linux. Appears to work fine on Windows. Set MWF_DATADIR to the directory path to load assets from. If assets.zip is in that directory, it will be loaded. Otherwise, raw assets will be loaded directly.

Please try loading with and without zip file.

File: Memes With Friends/Memes With Friends/main.cpp [hunk 1]
```diff
@@ -73,9 +79,27 @@ int main(void)
 		return -1;
 	}
 
+	// std::string from NULL char * is undefined behavior and causes crashing
+	// to resolve, we will try to grab the environment variable into datadir_raw
+	// if it's NULL or blank, initialize with contents of DATADIR
+	char *datadir_raw = std::getenv("MWF_DATADIR");
+	if (datadir_raw == NULL || std::strlen(datadir_raw) == 0) {
```
Comment (EliteTK) [inline Memes With Friends/Memes With Friends/main.cpp]: std::strlen needs \<cstring> (at least with GNU libstdc++)

File: Memes With Friends/Memes With Friends/main.cpp [hunk 2]
```diff
@@ -73,9 +79,27 @@ int main(void)
 		return -1;
 	}
 
+	// std::string from NULL char * is undefined behavior and causes crashing
+	// to resolve, we will try to grab the environment variable into datadir_raw
+	// if it's NULL or blank, initialize with contents of DATADIR
+	char *datadir_raw = std::getenv("MWF_DATADIR");
```
Comment (EliteTK) [inline Memes With Friends/Memes With Friends/main.cpp]:
```
main.cpp: In function ‘int main()’:
main.cpp:21:17: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DATADIR "./"
                 ^
main.cpp:88:17: note: in expansion of macro ‘DATADIR’
   datadir_raw = DATADIR;
                 ^~~~~~~
```

Needs to be "const char *" (pointer to const char).

Comment (dean1012) [inline Memes With Friends/Memes With Friends/main.cpp]: To be clear, are you saying I need to cast the constant to (const char *) or something else?

Comment (EliteTK) [inline Memes With Friends/Memes With Friends/main.cpp]: No, the variable used to store it should be "const char *".

Comment (dean1012) [inline Memes With Friends/Memes With Friends/main.cpp]: Ok so replace #DEFINE DATADIR with const char *datadir correct?

Comment (EliteTK) [inline Memes With Friends/Memes With Friends/main.cpp]: No, replace `char *datadir_raw = std::getenv("MWF_DATADIR");` with `const char *datadir_raw = ...;`.

**Comment by dean1012** [comment]
@EliteTK Please take a look at this again and make sure it works ok before I merge.

**Comment by EliteTK** [comment]
Compiles just fine.

**Comment by dean1012** [comment]
@EliteTK Thanks. Merging then to get one of these out the door.

**Pull request merged**